### DISCUSSION TITLE
ShowDialog and Result

### DISCUSSION THREAD

**Discussion by michalss** [AvaloniaUI/Avalonia#15228] [General]
Hi,

Can anyone please show me how to make sure i get own model from ShowDialog please? Using CommunityToolkit and MVVM. Problem is that result is always null and i dont know how to make sure i get result need from the Window?

```csharp
            var result =  await dlg.ShowDialog<Modd_Task>(App.MainWindow);
            var ssds = result;
```

**Comment by thevortexcloud** [comment]
You need to return a result if you are going to do it that way.

https://docs.avaloniaui.net/docs/reference/controls/window#show-a-window-as-a-dialog

**Reply by michalss** [reply]
thx i missed that ref. Hmm what would be best to do it from ViewModel pls ?

**Reply by timunie** [reply]
https://github.com/AvaloniaUI/Avalonia.Samples?tab=readme-ov-file#view-interaction-samples
